File: BubbleSort.pl
#! usr/bin/perl
print STDOUT "Enter a sequence of numbers to be sorted";
$Numbers = <>;
chop ($Numbers);
print "Hi: The Numbers are =$Numbers= \n\n";



@Nums = split(/\s+/, $Numbers);

for ($i=0;$i<=$#Nums;$i++) {

	# Bubble up;
	$Max_Array_Size = $#Nums +1 - $i ;
	# print "i=$i= Max_Array_size=$Max_Array_Size=  i=$i= tot=$#Nums=\n";
	&BubbleUp($Max_Array_Size) ;

}

print "Idrees sorted list is @Nums\n\n";



#Given an array of size N; make array[N-1] have the max

sub BubbleUp 
{

   local ($ArraySize) = @_;

    print "Idrees BubbleUp:: ArraySize=$ArraySize=\n";

    $PostnOfMax = $ArraySize-1;

    for ($k=0;$k<$ArraySize;$k++) {

	if ($Nums[$k] > $Nums[$PostnOfMax]) { 
		$temp = $Nums[$PostnOfMax] ;

		# Swap
		$Nums[$PostnOfMax] = $Nums[$k] ; 
		$Nums[$k] = $temp ;

	}



    }



}

File: BubbleSort.c
#include <stdio.h>
#include <stdlib.h>


int Array [] = {70, 64 ,2 ,10, 25, 75, 69, 79, 22, 34};

void PrintNumbers()
{
	int i ;
	
	printf ("Numbers = ");
	for (i=0;i<sizeof(Array)/sizeof(int);i++) {
		printf("%d ", Array[i]);
	}
	printf("\n");

}



void main()
{

	PrintNumbers();
	
	int i, j , ArraySize, IndexOfMax, temp;
	
	ArraySize = sizeof(Array) / sizeof(int);
	
	
	for (i=ArraySize-1;i>=0;i--){
		
		IndexOfMax = i ;
		// Find the max
		for (j=i;j>=0;j--) {
			if (Array[j] > Array[IndexOfMax]) {
				IndexOfMax = j ;
			}			
		}
		
		// swap
		
		temp = Array[i];
		Array[i] = Array[IndexOfMax];
		Array[IndexOfMax] = temp ;
		PrintNumbers() ;
		
		
		
	}


}